### ISSUE TITLE
Filterable Multiselect focus outline on wrong node

### ISSUE THREAD

**Issue by wkeese** [carbon-design-system/carbon#7535]
When a filterable multiselect has some options selected, clicking on it's `<input>` puts focus outline on wrong node:

<img width="321" alt="Screen Shot 2021-01-12 at 22 05 18" src="https://user-images.githubusercontent.com/69599/104319050-a9727900-5523-11eb-9ff0-893e849bf8d7.png">

## Environment

> Operating system

MacOS

> Browser

Chrome

## Detailed description

> What version of the Carbon Design System are you using?

10.26

> What did you expect to happen?

Focus outline around whole control.

> What happened instead?

Focus outline around `<input>`

## Test case

See https://react.carbondesignsystem.com/?path=/story/multiselect--filterable or https://react.carbondesignsystem.com/?path=/story/multiselect--filterable

Note that clicking more to the left will put the focus outline around the whole control:

<img width="342" alt="Screen Shot 2021-01-12 at 22 19 22" src="https://user-images.githubusercontent.com/69599/104319508-50571500-5524-11eb-9ccb-67ae50e1831f.png">

Presumably a regression from #4721.

**Comment by dakahn** [comment]
@wkeese I can't recreate this on Windows in either Chrome or Firefox using our Storybook. Do you still see this error?

**Comment by wkeese** [comment]
Yes, I do. Just retested on Chrome on Windows 10.   Probably, you forgot to select some options in the dropdown first?  Try using TAB and SHIFT-TAB to go back and forth between the (2x) button and the `<input>`.

<img width="628" alt="Screen Shot 2021-01-16 at 13 42 16" src="https://user-images.githubusercontent.com/69599/104797356-c7c8c500-5800-11eb-85b5-956e9975b0dd.png">

Also, note that this is about the *filterable* multiselect.

**Comment by dakahn** [comment]
@wkeese thanks I see it now, thanks for your patience. 

@carbon-design-system/design while it might look a little questionable I don't actually have a problem with this behavior from an accessibility stand point. The text input is receiving focus correctly. In fact an argument could be made that this behavior _helps_ magnification users better determine the actual input on the form as distinct from the button to clear the selections since visually we combine a listbox and a button here.

**Comment by wkeese** [comment]
I agree you can debate which focus outline is better from an accessibility standpoint, but I think you need to pick one way or the other.  Right now tabbing into or clicking on the `<input>` puts the focus outline on the `<input>`:

<img width="344" alt="Screen Shot 2021-01-22 at 8 13 37" src="https://user-images.githubusercontent.com/69599/105424016-dc3a1100-5c89-11eb-8e33-2758774fcdc3.png">

But clicking on the upper left (when options are selected) puts the focus outline on the whole control:

<img width="341" alt="Screen Shot 2021-01-22 at 8 09 20" src="https://user-images.githubusercontent.com/69599/105423824-967d4880-5c89-11eb-8181-1bad4edc25b8.png">

The other (very related) issue is that there's actually an extra tab stop on the whole control: clicking on the upper left and then pressing the tab key goes to the (2 x) button, and then another tab goes to the `<input>`... i.e. there are 3 tab stops where there should be 2.

(Both these screenshots are from React on Chrome/mac but I assume it's a universal behavior.)

PPS: I would also question the slightly different background colors for the left side vs. the input.

**Comment by wkeese** [comment]
Looks like you fixed this.  Judging from https://react.carbondesignsystem.com/?path=/story/components-multiselect--filterable the focus outline is always around the whole control (except when the (2x) button is focused, in which case the focus outline is just on that button).
